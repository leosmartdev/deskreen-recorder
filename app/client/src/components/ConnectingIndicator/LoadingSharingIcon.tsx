import React, { useContext } from 'react';
import { Icon } from '@blueprintjs/core';
import { Col, Row } from 'react-flexbox-grid';
import PropagateLoader from 'react-spinners/PropagateLoader';
import { AppContext } from '../../providers/AppContextProvider';

const Fade = require('react-reveal/Fade');

interface SelectSharingIconProps {
  loadingSharingIconType: LoadingSharingIconType;
  isShownLoadingSharingIcon: boolean;
}

function LoadingSharingIcon(props: SelectSharingIconProps) {
  const { isDarkTheme } = useContext(AppContext);

  const {
    loadingSharingIconType: selectingSharingIconType,
    isShownLoadingSharingIcon: isShownSelectingSharingIcon,
  } = props;

  return (
    <Row
      center="xs"
      top="xs"
      style={{
        height: '100%',
        width: '100%',
        marginRight: '0px',
        marginLeft: '0px',
      }}
    >
      <Col>
        <Row
          style={{
            width: '100%',
            height: '50px',
          }}
          center="xs"
        >
          <Col xs={8} md={4}>
            <PropagateLoader
              loading
              size={18}
              color={isDarkTheme ? '#BFCCD6' : '#5C7080'}
            />
          </Col>
        </Row>
        <Row center="xs">
          <Col>
            <Fade opposite when={isShownSelectingSharingIcon} duration={500}>
              <Icon
                icon={selectingSharingIconType}
                iconSize={60}
                color={isDarkTheme ? '#BFCCD6' : '#5C7080'}
              />
            </Fade>
          </Col>
        </Row>
      </Col>
    </Row>
  );
}

export default LoadingSharingIcon;
