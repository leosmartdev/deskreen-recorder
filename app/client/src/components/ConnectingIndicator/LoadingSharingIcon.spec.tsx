import React from 'react';
import renderer from 'react-test-renderer';
import LoadingSharingIcon from './LoadingSharingIcon';

jest.useFakeTimers();

it('should match exact snapshot', () => {
  const subject = renderer.create(
    <>
      <LoadingSharingIcon
        loadingSharingIconType="desktop"
        isShownLoadingSharingIcon
      />
    </>
  );

  expect(subject).toMatchSnapshot();
});
