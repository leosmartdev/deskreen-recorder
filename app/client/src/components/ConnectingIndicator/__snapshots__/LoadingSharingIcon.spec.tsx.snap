// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match exact snapshot 1`] = `
<div
  className="row center-xs top-xs"
  style={
    Object {
      "height": "100%",
      "marginLeft": "0px",
      "marginRight": "0px",
      "width": "100%",
    }
  }
>
  <div
    className=""
  >
    <div
      className="row center-xs"
      style={
        Object {
          "height": "50px",
          "width": "100%",
        }
      }
    >
      <div
        className="col-xs-8 col-md-4"
      >
        <div
          className="css-9ebb9a"
        >
          <div
            className="css-1nih24x"
          />
          <div
            className="css-182ruc5"
          />
          <div
            className="css-fapzgm"
          />
          <div
            className="css-fqj3wr"
          />
          <div
            className="css-srr2wk"
          />
          <div
            className="css-1netspe"
          />
        </div>
      </div>
    </div>
    <div
      className="row center-xs"
    >
      <div
        className=""
      >
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <span
            className="bp3-icon bp3-icon-desktop"
            icon="desktop"
          >
            <svg
              data-icon="desktop"
              fill="#5C7080"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <desc>
                desktop
              </desc>
              <path
                d="M19 0H1C.45 0 0 .45 0 1v13c0 .55.45 1 1 1h5.67l-.5 3H5c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1h-1.17l-.5-3H19c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1zm-1 13H2V2h16v11z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
